packages/flame/lib/src/game/mixins/has_fps.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:math';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/scheduler.dart';
+
+import '../../../game.dart';
+
+const _maxFrames = 60.0;
+const frameInterval =
+    Duration(microseconds: Duration.microsecondsPerSecond ~/ _maxFrames);
+
+@Deprecated(
+  'This will be removed in favor of `HasFPS` in v1.3.0',
+)
+typedef FPSCounter = HasFPS;
+
+mixin HasFPS on Game {
+  List<FrameTiming> _previousTimings = [];
+
+  @override
+  void onTimingsCallback(List<FrameTiming> timings) =>
+      _previousTimings = timings;
+
+  /// Returns the FPS based on the frame times from [onTimingsCallback].
+  double fps([
+    @Deprecated('This will be removed in v1.3.0') int average = 1,
+  ]) {
+    return min(
+      Duration.microsecondsPerSecond /
+          (_previousTimings.map((t) => t.totalSpan.inMicroseconds).sum /
+              _previousTimings.length),
+      _maxFrames,
```

Comment:
I wonder why this upper limit was added here?
Flutter can go above 60fps on certain devices (https://flutter.cn/perf/ui-performance#:~:text=Flutter%20aims%20to%20provide%2060,to%20render%20approximately%20every%2016ms.)

Reply:
Because the waiting time will be done outside of the frame ticks, so it can report 400 fps here when it is idle. But you're right, the upper limit should be the device limit, which I think is possible to get.

Reply:
What I think is that it measures the time for the frames, but if it has "time left" it will not be in a frame timing. It seems to make sense at least since it produces a sensible result when the frame rate drops.

Reply:
Oh, I think I see what you mean. If frames fire every 16ms (i.e. 60fps), but each frame takes only 1ms to build, then the timings will report [1ms, 1ms, 1ms, ...], which will look like it's 1000fps while in fact it's still only 60fps, because in reality there are 15ms pauses between each 1ms.

Reply:
Yeah that's true, I think we should try to do an FPS component again, maybe the same faulty math was used when that was tried last time.